File: src/sranuluge/python/find-duplicates/method2.py
properties_list_1 = ["a", "b", "c", "d", "e"]
properties_list_2 = ["f", "g", "w", "i", "h"]
#Using map
map_values = {}
def is_list_contain_duplicates_2(arr1, arr2):
    for item in arr1:
        if item not in map_values:
            map_values[item] = True
    
    for nested_item in arr2:
        if nested_item in map_values:
            return map_values[nested_item]
    return False

# O(a + b) when comes to the time complexity => good
# O(a) Space complexity bad => creating a objet

print(f'Method-02: {is_list_contain_duplicates_2(properties_list_1, properties_list_2)}')

File: src/sranuluge/javascript/find-duplicates/method2.js
import { array1, array2 } from "./data";

let map = {};
function isContainDuplicate() {
  for (const [i, v] of array1.entries()) {
    if (!map[v]) {
      map[v] = true;
    }
  }

  for (const [j, q] of array2.entries()) {
    if (map[q]) {
      return map[q];
    }
  }
  return false;  
}

//o(a + b) when comes to the time complexity good
//o(a) Space complexity bad => creating a objet
console.log(isContainDuplicate(array1, array2));
